Mage.Sets/src/mage/cards/x/XyrisTheWrithingStorm.java

```diff
@@ -0,0 +1,157 @@
+import mage.game.permanent.token.SnakeToken;
+import mage.game.permanent.token.Token;
+import mage.players.Player;
+import mage.watchers.common.CardsDrawnDuringDrawStepWatcher;
+
+import java.util.UUID;
+
+/**
+ * @author AsterAether
+ */
+public final class XyrisTheWrithingStorm extends CardImpl {
+
+    public XyrisTheWrithingStorm(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}{U}{R}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SNAKE);
+        this.subtype.add(SubType.LEVIATHAN);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(5);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Whenever an opponent draws a card except the first one they draw in each of their draw steps, create a 1/1 green Snake creature token.
+        this.addAbility(new XyrisTheWrithingStormDrawAbility(), new CardsDrawnDuringDrawStepWatcher());
+        // Whenever Xyris, the Writhing Storm deals combat damage to a player, you and that player each draw that many cards.
+        this.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(new XyrisTheWrithingStormCombatDamageEffect(), false, true));
+    }
+
+    private XyrisTheWrithingStorm(final XyrisTheWrithingStorm card) {
+        super(card);
+    }
+
+    @Override
+    public XyrisTheWrithingStorm copy() {
+        return new XyrisTheWrithingStorm(this);
+    }
+}
+
+class XyrisTheWrithingStormDrawAbility extends TriggeredAbilityImpl  {
+
+    public XyrisTheWrithingStormDrawAbility() {
+        super(Zone.BATTLEFIELD, new XyrisTheWrithingStormTokenEffect(), false);
+    }
+
+    public XyrisTheWrithingStormDrawAbility(final XyrisTheWrithingStormDrawAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DREW_CARD;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (game.getPlayer(this.getControllerId()).hasOpponent(event.getPlayerId(), game)) {
+            if (game.isActivePlayer(event.getPlayerId())
+                    && game.getPhase().getStep().getType() == PhaseStep.DRAW) {
+                CardsDrawnDuringDrawStepWatcher watcher = game.getState().getWatcher(CardsDrawnDuringDrawStepWatcher.class);
+                if (watcher != null && watcher.getAmountCardsDrawn(event.getPlayerId()) > 1) {
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public TriggeredAbility copy() {
+        return new XyrisTheWrithingStormDrawAbility(this);
+    }
+
+    @Override
+    public String getRule() {
+        return "Whenever an opponent draws a card except the first one they draw in each of their draw steps" + super.getRule();
```

Comment:
You can put the whole ability text here instead of calling the super but if you don't you should add a comma after "steps"